src/timely-util/src/order.rs

```diff
@@ -150,6 +164,33 @@ impl Extrema for i32 {
     }
 }
 
+impl Extrema for Uuid {
+    fn minimum() -> Self {
+        Self::nil()
+    }
+    fn maximum() -> Self {
+        Self::from_bytes([0xff; 16])
```

Comment:
I think it's the other way around. Inherent methods shadow trait methods and if you want to use the trait one you can do `<Uuid as Ord>::max` or something like that.  https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=75ab9ac3c36ddaa8bd000a1d94ad79e4

Reply:
hmm that's what I expected but rust-analyzer seems to be defaulting to the Ord impl instead. I'll try to compile regardless once I cleanup some of the other things

Reply:
Turns out this is because we're using `uuid` `1.2.2` and in that version the `max` inherent function was behind a feature flag https://docs.rs/uuid/1.2.2/src/uuid/builder.rs.html#98

Let's update the uuid crate since we're here

Reply:
Still haven't gotten this to work using `Self::max()` -- it's not just rust-analyzer, somehow the `core::cmp` method is shadowing the inherent method:
```
error[E0061]: this function takes 2 arguments but 0 arguments were supplied
   --> src/timely-util/src/order.rs:173:9
    |
173 |         Self::max()
    |         ^^^^^^^^^-- two arguments of type `uuid::Uuid` and `uuid::Uuid` are missing
    |
note: method defined here
   --> /Users/roshan/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/cmp.rs:808:8
    |
808 |     fn max(self, other: Self) -> Self
    |        ^^^
help: provide the arguments
    |
173 |         Self::max(/* uuid::Uuid */, /* uuid::Uuid */)
    |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

For more information about this error, try `rustc --explain E0061`.
```

Reply:
ah good call! will update the UUID version, thanks!